cmd/api/main.go

```diff
@@ -169,7 +170,50 @@ func createBook(w http.ResponseWriter, r *http.Request) {
 
 /* Returns a list of the stored books. */
 func getBooks(w http.ResponseWriter, r *http.Request) {
-	returnedBooks, err := listBooks()
+
+	err := r.ParseForm() //Verify errors in request query params and populates r.Form with its values.
+	if err != nil {
+		log.Println(err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	//Adapt query params:
+	name := r.FormValue("name")
+	if name == "" {
+		name = "%"
+	}
+
+	var minPrice32 float32
+	minPriceStr := r.FormValue("min_price")
+	if minPriceStr != "" {
+		minPrice64, err := strconv.ParseFloat(minPriceStr, 32)
+		if err != nil {
+			log.Println(err)
+			w.WriteHeader(http.StatusInternalServerError)
```

Comment:
This is probably a user error, if the user passes an invalid float string this error would occur.